### PR TITLE
Merge csi v1.0 to master

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@rootfs  too :)

**Comment by ShyamsundarR** [comment]
@humblec this seems to be merging commits from master to csi-v1.0 whereas the title of the PR suggests otherwise, can you clarify?

**Comment by ShyamsundarR** [comment]
I had done a short excercise yesterday to see what patches are part of csi-v1.0 and not master and had requested backport (or forward ports) of the following PRs,
- https://github.com/ceph/ceph-csi/pull/334
- https://github.com/ceph/ceph-csi/pull/333

Leaving these here for consideration.

**Comment by humblec** [comment]
@ShyamsundarR how it looks now ?

**Comment by Madhu-1** [comment]
@ShyamsundarR changes looks okay?

**Comment by Madhu-1** [comment]
we should take this PR in on top priority.

File: Makefile [hunk 1]
```diff
@@ -25,6 +25,9 @@ CEPHFS_IMAGE_VERSION=$(if $(ENV_CEPHFS_IMAGE_VERSION),$(ENV_CEPHFS_IMAGE_VERSION
 CSI_IMAGE_NAME?=quay.io/cephcsi/cephcsi
 CSI_IMAGE_VERSION?=v1.0.0
 
+CSI_IMAGE_NAME?=quay.io/cephcsi/cephcsi
```
Comment (Madhu-1) [inline Makefile]: @humblec looks like we have duplicate lines in Makefile

Comment (humblec) [reply Makefile]: @Madhu-1 checking

Comment (humblec) [reply Makefile]: @Madhu-1 corrected. Thanks!

**Comment by ShyamsundarR** [comment]
> @ShyamsundarR how it looks now ? Thanks

Good, changes are now limited to the 2 PRs that were outstanding as in earlier comment. Request you to update those PRs stating that the code is now ported over to master as well, so that owners do not spend cycles porting the same.

I am good with the changes here.

**Comment by humblec** [comment]
FYI , Notified those PRs. Thanks.. This is good to go then!

**Comment by Madhu-1** [comment]
why Travis CI is still pending :(

**Comment by ShyamsundarR** [comment]
This brings in a series of 17 commits, instead of just backporting the 2 PRs, isn't there a better way to get this done? For example @phlogistonjohn's commits are being replayed here, whereas these are already merged into master.

Further there seems to be a rebase due to conflict error shown in the PR, which may prevent the auto merge anyway.

**Comment by humblec** [comment]
Actually there are `9` commits in the PRs mentioned above,  Here its `16` , out of these `16` there is a merge commit and `2` merge conflict correct commits, so its just `4` commits difference.

**Comment by j-griffith** [comment]
Could consider a retriggering the Travis job, not something I have perms for.

**Comment by humblec** [comment]
@j-griffith true, I think we can manually merge this PR. What do you guys think ? We need to get out of this mess at the earliest.

**Comment by humblec** [comment]
@j-griffith retriggered, lets wait. Thanks!

**Pull request merged**